src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,68 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onSearchQuery: React.Dispatch<React.SetStateAction<string>>,
+  filter: string,
+  onFilter: React.Dispatch<React.SetStateAction<string>>
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onSearchQuery,
+  filter,
+  onFilter,
+}) => {
+  const [query, setQuery] = useState('');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filter}
+            onChange={event => {
+              onFilter(event.target.value);
+            }}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={event => {
+            setQuery(event.target.value);
+            onSearchQuery(event.target.value);
+          }}
```

Comment:
Create a separate handler for this function, and you can destructing the `value` from the `event.target` to avoid repeating them